### PR TITLE
add map.getMaxBounds method

### PR DESCRIPTION
**Pull request by andrewharvey**
## Launch Checklist

part of #4029 

Thanks to @lamuertepeluda for the inspiration in https://github.com/mapbox/mapbox-gl-js/issues/4029#issuecomment-294713995

 - [x] briefly describe the changes in this PR
 - [x] write tests for all new functionality
 - [x] document any changes to public APIs
 - [ ] post benchmark scores
 - [ ] manually test the debug page

### CODE REVIEW

File: src/ui/map.js [hunk 1]
```diff
@@ -407,6 +407,22 @@ class Map extends Camera {
...
 
     /**
+     * Gets the map's geographical bounds.
+     *
+     * Returns the LngLatBounds by which pan and zoom operations on the map are constrained.
+     *
+     * @returns {LngLatBounds | null} The maximum bounds the map is constrained to, or `null` if none set.
+     */
+    getMaxBounds () {
+        if (!(!Array.isArray(this.transform.latRange) || this.transform.latRange.length === 0) &&
```
Comment (mollymerp) [inline src/ui/map.js]: I find this a little hard to read  😕 can we invert the check ( `if (this.transform.latRange && this.transform.latRange.length == 2) && ... else { return null; } `) for clarity?

File: src/ui/map.js [hunk 2]
```diff
@@ -434,6 +450,7 @@ class Map extends Camera {
         return this;
 
     }
+
```
Comment (mollymerp) [inline src/ui/map.js]: accidental line? 🙃

Comment (andrewharvey) [inline src/ui/map.js]: everywhere else had a whitespace between methods, so I added one here to be consistent.

**Comment by andrewharvey** [comment]
should be good now

**Pull request merged**